docker/zap-api-scan.py

```diff
@@ -446,6 +438,14 @@ def main(argv):
                     # Choose the first one - will be striping off the path below
                     target = urls[0]
                     logging.debug('Using target from imported file: {0}'.format(target))
+
```

Comment:
I dont think that moving this code has the effect you want. 
The `res = zap.openapi.import_file(target_file)` line does the actual importing, and that does not support the host override option.
The openapi add-on will also have to change in order to override the hosts for local files.